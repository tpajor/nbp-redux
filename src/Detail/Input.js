import React from 'react';

import './Input.css';

export default class Input extends React.Component {
  state = { 
    numberOfData: '',
  };

  handleSubmit = event => {
    event.preventDefault();
    const value = parseInt(this.state.numberOfData, 10);
    this.props.getDataForChart(value);
  };

  render() {
    const { numberOfData } = this.state;
    const { inputError, errorMessage } = this.props;
    return (
      <form onSubmit={this.handleSubmit} className={'Form'}>
        <div className={'form-group col-2 Input'}>
          <input className="form-control-plaintext"
            type="text" 
            value={this.props.code.toUpperCase()}
            readOnly
          />
        </div>
        <div className={'form-group col-6 Form'}>
          <input className={`form-control ${inputError ? 'is-invalid': ''}`}
            type="text" 
            value={numberOfData}
            onChange={event => this.setState({ numberOfData: event.target.value })}
            placeholder="Wpisz liczbę ostatnich kursów"
          />
          {(inputError) ?
            <p style={{color: 'red'}}>{errorMessage}</p> :
            ''
          }
        </div>
        <button type="submit" className={'btn btn-primary col-4 SearchButton'} disabled={this.props.code === ''}>Pokaż</button>
      </form> 
    );
  }
};