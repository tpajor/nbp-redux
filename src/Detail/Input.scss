.Input {
  display: inline-block;
  width: 100%;
  vertical-align: top;
  margin: 20px 0 20px 0;
}

.Form {
  margin: 20px 0 20px 0;
  display: inline-block;
  vertical-align: middle;
  width: 100%;
  position: relative;
}

.SearchButton {
  position: absolute;
  margin-top: 20px;
}
