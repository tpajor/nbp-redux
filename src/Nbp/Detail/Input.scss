.Input {
  display: inline-block;
  width: 100%;
  vertical-align: top;
  margin: 20px 0 20px 0;
}

.Form {
  margin: 20px 0 20px 0;
  display: inline-block;
  vertical-align: middle;
  width: 100%;
  position: relative;
}

.InputButton {
  position: absolute;
  margin-top: 20px;
}
