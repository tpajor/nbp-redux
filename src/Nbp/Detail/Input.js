import React from 'react';
import PropTypes from 'prop-types';

import './Input.css';

export default class Input extends React.Component {
  state = { 
    numberOfData: '',
  };

  handleSubmit = event => {
    event.preventDefault();
    const value = parseInt(this.state.numberOfData, 10);
    this.props.getInputedNumberOfLastRates(value, this.props.currencyCodeToViewInDetail);
  };

  render() {
    const { numberOfData } = this.state;
    const { detailsInputError, detailsInputErrorMessage, currencyCodeToViewInDetail, input } = this.props;
    return (
      <form onSubmit={this.handleSubmit} className={'Form'}>
        <div className={'form-group col-2 Input'}>
          <input className="form-control-plaintext"
            type="text" 
            value={currencyCodeToViewInDetail.toUpperCase()}
            readOnly
          />
        </div>
        <div className={'form-group col-6 Form'}>
          <input className={`form-control ${detailsInputError ? 'is-invalid': ''}`}
            type="text" 
            value={currencyCodeToViewInDetail ? numberOfData : ''}
            onChange={event => this.setState({ numberOfData: event.target.value })}
            placeholder="Wpisz liczbę ostatnich kursów"
          />
          {(detailsInputError) ?
            <p style={{color: 'red'}}>{detailsInputErrorMessage}</p> :
            ''
          }
        </div>
        <button type="submit" className={'btn btn-primary col-4 InputButton'} disabled={currencyCodeToViewInDetail === ''}>Pokaż</button>
      </form> 
    );
  }
};

Input.propTypes = {
  currencyCodeToViewInDetail: PropTypes.string,
  getInputedNumberOfLastRates: PropTypes.func,
  detailsInputError: PropTypes.bool,
  detailsInputErrorMessage: PropTypes.string,
};