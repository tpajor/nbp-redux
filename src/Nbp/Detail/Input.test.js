import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Input from './Input';


Enzyme.configure({ adapter: new Adapter() });

const initialState = { 
  numberOfData: '',
};

function setup() {
  const props = {
    currencyCodeToViewInDetail: 'USD',
    detailsInputError: false,
    detailsInputErrorMessage: '',
    getInputedNumberOfLastRates: jest.fn(),
  };

  const enzymeWrapper = shallow(<Input {...props} />);

  return {
    enzymeWrapper,
    props
  };
};

describe('Input component', () => {
  it('should render self and subcomponents with their props', () => {
    const { enzymeWrapper, props } = setup();
    enzymeWrapper.setState({ ...initialState });

    expect(enzymeWrapper.length).toBe(1);

    expect(enzymeWrapper.find('form').length).toBe(1);
    expect(enzymeWrapper.find('form').hasClass('Form')).toBe(true);

    expect(enzymeWrapper.find('div').length).toBe(2);
    expect(enzymeWrapper.find('div').first().hasClass('form-group col-2 Input')).toBe(true);
    expect(enzymeWrapper.find('div').last().hasClass('form-group col-6 Form')).toBe(true);

    expect(enzymeWrapper.find('input').length).toBe(2);
    expect(enzymeWrapper.find('input').first().hasClass('form-control-plaintext')).toBe(true);
    expect(enzymeWrapper.find('input').first().props().value).toEqual('USD');
    expect(enzymeWrapper.find('input').last().hasClass('form-control ')).toBe(true);
    enzymeWrapper.setProps({ ...props, detailsInputError: true });
    expect(enzymeWrapper.find('input').last().hasClass('form-control is-invalid')).toBe(true);
    enzymeWrapper.setProps({ ...props });
    enzymeWrapper.setState({ numberOfData: 234 })
    expect(enzymeWrapper.find('input').last().props().value).toEqual(234);
    enzymeWrapper.setProps({ ...props, currencyCodeToViewInDetail: '' });
    expect(enzymeWrapper.find('input').last().props().value).toEqual('');

    expect(enzymeWrapper.find('p').length).toBe(0);
    enzymeWrapper.setProps({ ...props, detailsInputError: true, detailsInputErrorMessage: 'error' });
    expect(enzymeWrapper.find('p').length).toBe(1);
    expect(enzymeWrapper.find('p').text()).toEqual('error');
    enzymeWrapper.setProps({ ...props });

    expect(enzymeWrapper.find('button').length).toBe(1);
    expect(enzymeWrapper.find('button').hasClass('btn btn-primary col-4 InputButton')).toBe(true);
  });

  it('should render correctly', () => {
    const { enzymeWrapper } = setup();
    
    expect(enzymeWrapper).toMatchSnapshot();
  });

  it('should change state on second input change', () => {
    const { enzymeWrapper } = setup();
    const input = enzymeWrapper.find('input').last();
    expect(enzymeWrapper.state(['numberOfData'])).toEqual('');
    input.simulate('change', { target: { value: 'u' } });
    expect(enzymeWrapper.state(['numberOfData'])).toEqual('u');
  });

  it('should call getInputedNumberOfLastRates on form submit', () => {
    const { enzymeWrapper, props } = setup();
    const form = enzymeWrapper.find('form');
    form.simulate('submit', { preventDefault: () => {} });
    expect(props.getInputedNumberOfLastRates.mock.calls.length).toBe(1);
  });
});