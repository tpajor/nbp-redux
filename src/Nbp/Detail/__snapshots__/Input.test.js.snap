// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    currencyCodeToViewInDetail="USD"
    detailsInputError={false}
    detailsInputErrorMessage=""
    getInputedNumberOfLastRates={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="form-group col-2 Input"
>
          <input
                    className="form-control-plaintext"
                    readOnly={true}
                    type="text"
                    value="USD"
          />
</div>,
        <div
          className="form-group col-6 Form"
>
          <input
                    className="form-control "
                    onChange={[Function]}
                    placeholder="Wpisz liczbę ostatnich kursów"
                    type="text"
                    value=""
          />
          
</div>,
        <button
          className="btn btn-primary col-4 InputButton"
          disabled={false}
          type="submit"
>
          Pokaż
</button>,
      ],
      "className": "Form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="form-control-plaintext"
            readOnly={true}
            type="text"
            value="USD"
/>,
          "className": "form-group col-2 Input",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control-plaintext",
            "readOnly": true,
            "type": "text",
            "value": "USD",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="form-control "
              onChange={[Function]}
              placeholder="Wpisz liczbę ostatnich kursów"
              type="text"
              value=""
/>,
            "",
          ],
          "className": "form-group col-6 Form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control ",
              "onChange": [Function],
              "placeholder": "Wpisz liczbę ostatnich kursów",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Pokaż",
          "className": "btn btn-primary col-4 InputButton",
          "disabled": false,
          "type": "submit",
        },
        "ref": null,
        "rendered": "Pokaż",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="form-group col-2 Input"
>
            <input
                        className="form-control-plaintext"
                        readOnly={true}
                        type="text"
                        value="USD"
            />
</div>,
          <div
            className="form-group col-6 Form"
>
            <input
                        className="form-control "
                        onChange={[Function]}
                        placeholder="Wpisz liczbę ostatnich kursów"
                        type="text"
                        value=""
            />
            
</div>,
          <button
            className="btn btn-primary col-4 InputButton"
            disabled={false}
            type="submit"
>
            Pokaż
</button>,
        ],
        "className": "Form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="form-control-plaintext"
              readOnly={true}
              type="text"
              value="USD"
/>,
            "className": "form-group col-2 Input",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control-plaintext",
              "readOnly": true,
              "type": "text",
              "value": "USD",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="form-control "
                onChange={[Function]}
                placeholder="Wpisz liczbę ostatnich kursów"
                type="text"
                value=""
/>,
              "",
            ],
            "className": "form-group col-6 Form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control ",
                "onChange": [Function],
                "placeholder": "Wpisz liczbę ostatnich kursów",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Pokaż",
            "className": "btn btn-primary col-4 InputButton",
            "disabled": false,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Pokaż",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
